cloud/terraform/terraform_controller.py

```diff
@@ -0,0 +1,122 @@
+import os
+
+
+class TerraformController:
+    def __init__(self, cloud):
+        self.cloud = cloud
+
+        #TODO: where is this executed from? change path?
+        if os.path.exists("main.tf"):
+            os.remove("main.tf")
+        header = """
+terraform {
+    required_providers {
+        aws = {
+        source  = "hashicorp/aws"
+        version = "~> 3.27"
+        }
+    }
+
+    required_version = ">= 0.14.9"
+}
+"""
+        self._add_to_config_file(header)
+
+    def _add_to_config_file(self, addition):
+        with open("main.tf", "a") as file:
+            file.write(addition)
+        
+    def _string_in_config(self, test_string):
```

Comment:
I suggest renaming to `_search_string_in_file` and pass the file path as argument. Then this function can be moved to utils.